src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2011-2013, CloudBees, Inc., Stephen Connolly.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.hudson.plugins.folder.computed;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.TopLevelItem;
+import hudson.tasks.LogRotator;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.GregorianCalendar;
+import java.util.Iterator;
+import java.util.List;
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.FINER;
+import java.util.logging.Logger;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * The default {@link OrphanedItemStrategy}.
+ * Trims dead items by the # of days or the # of jobs much like {@link LogRotator}.
+ *
+ * @author Stephen Connolly
+ */
+public class DefaultOrphanedItemStrategy extends OrphanedItemStrategy {
+
+    /**
+     * Our logger
+     */
+    private static final Logger LOGGER = Logger.getLogger(DefaultOrphanedItemStrategy.class.getName());
+
+    /**
+     * Should old branches be removed at all
+     */
+    private final boolean pruneDeadBranches;
```

Comment:
Yes, using `addCompatibilityAlias`; see `PeriodicFolderTrigger` for an example. The difficulty with `OrphanedItemStrategy` is that the former field referring to the former class also had a different name (`deadBranchStrategy`), and there is no simple way to rename fields. You can do it with some work with a `readResolve` method. I intended to put this off until there is agreement from @gusreiber on whether we even want to retain the same config UI as was previously used, or switch to a different style that would require a slightly more complex migration. Also until I find the time to write some `@LocalData` tests proving that the implementation works as intended.